R/combine_checkboxes.R

```diff
@@ -40,87 +41,84 @@
-  check_fields_exist(fields = field_names, expr = cols_exp)
+  # Save user cols to quo
+  cols_exp <- enquo(cols)
 
-  # Define values_to as the count of TRUEs/1s for the given checkbox field
-  # Assign TRUE if multiple selections made, and FALSE if one or zero made
-  data_tbl_mod <- data_tbl %>%
-    mutate(
-      !!values_to := case_when(
-        rowSums(select(., eval_tidy(cols_exp))) > 1 ~ TRUE,
-        TRUE ~ FALSE
-      )
-    )
+  # Evaluate the cols expression to get the selected column names
+  selected_cols <- names(eval_select(cols_exp, data = data_tbl))
+  check_fields_exist(fields = selected_cols, expr = cols_exp) # Check supplied fields exist
 
   # Get metadata reference table, check that chosen fields are checkboxes
-  metadata <- get_metadata_ref(data_tbl_mod, supertbl, tbl, field_names)
+  metadata_tbl <- supertbl$redcap_metadata[supertbl$redcap_form_name == tbl][[1]]
+  metadata_spec <- get_metadata_spec(metadata_tbl, selected_cols, names_prefix, names_suffix, names_sep)
 
-  # Replace TRUEs/1s with raw/label values from metadata
-  data_tbl_mod <- data_tbl_mod %>%
-    mutate(across(
-      field_names,
-      ~ replace_true(.x,
-        cur_column(),
-        metadata = metadata,
-        raw_or_label = raw_or_label
+  # Define .new_col as the count of TRUEs/1s for the given checkbox field
+  # Assign TRUE if multiple selections made, and FALSE if one or zero made
+  data_tbl_mod <- data_tbl
+  .new_col <- unique(metadata_spec$.new_value)
+
+  for (i in seq_along(.new_col)) {
+    cols_to_sum <- metadata_spec$field_name[metadata_spec$.new_value == .new_col[i]] # nolint: object_usage_linter
+
+    data_tbl_mod <- data_tbl_mod %>%
+      mutate(
+        !!.new_col[i] := case_when(
+          rowSums(select(., cols_to_sum)) > 1 ~ TRUE,
+          .default = FALSE
+        )
       )
-    ))
-
+  }
 
-  # Convert values_to from TRUE/FALSE to multi_value_label or identified single val
+  # Replace TRUEs/1s with raw/label values from metadata
   data_tbl_mod <- data_tbl_mod %>%
     mutate(
-      across(field_names, as.character), # enforce to character strings
-      across(!!values_to, ~ as.character(.))
-    ) %>%
-    mutate(
-      !!values_to := ifelse(!!sym(values_to) == "TRUE",
-        multi_value_label,
-        coalesce(!!!syms(field_names))
+      across(
+        selected_cols,
+        ~ replace_true(.x,
+          cur_column(),
+          metadata = metadata_spec,
+          raw_or_label = raw_or_label
+        )
       ),
-      !!values_to := ifelse(is.na(!!sym(values_to)),
-        values_fill,
-        !!sym(values_to)
-      )
-    ) %>%
-    mutate(
-      !!values_to := factor(!!sym(values_to),
-        levels = c(metadata[[raw_or_label]], multi_value_label, values_fill)
-      )
+      across(selected_cols, as.character) # enforce to character strings
     )
 
+  # Use the metadata_spec table to fill values in .new_col
+  data_tbl_mod <- reduce(.new_col, function(tbl, col_item) {
```

Comment:
Related to the idea above. Consider creating new columns independently and binding everything together rather than building the table incrementally and using `reduce`